src/net/socket_utils.zig

```diff
@@ -17,53 +17,42 @@ pub fn readSocket(
     exit: *const std.atomic.Value(bool),
     logger: Logger,
 ) !void {
-    //Performance out of the IO without poll
+    // Performance out of the IO without poll
     //  * block on the socket until it's readable
     //  * set the socket to non blocking
     //  * read until it fails
     //  * set it back to blocking before returning
 
-    const MAX_WAIT_NS = std.time.ns_per_ms; // 1ms
-
     while (!exit.load(.unordered)) {
         // init a new batch
-        var count: usize = 0;
-        const capacity = PACKETS_PER_BATCH;
-        var packet_batch = try std.ArrayList(Packet).initCapacity(
-            allocator,
-            capacity,
-        );
-        packet_batch.appendNTimesAssumeCapacity(Packet.default(), capacity);
+        var packet_batch = try std.ArrayList(Packet).initCapacity(allocator, PACKETS_PER_BATCH);
+        errdefer packet_batch.deinit();
 
         // NOTE: usually this would be null (ie, blocking)
-        // but in order to exit cleanly in tests - we set to 1 second
-        try socket.setReadTimeout(std.time.ms_per_s);
-        var timer = std.time.Timer.start() catch unreachable;
+        // but in order to exit cleanly in tests we set to 1 second
+        try socket.setReadTimeout(1 * std.time.ms_per_s);
 
         // recv packets into batch
         while (true) {
-            const n_packets_read = recvMmsg(socket, packet_batch.items[count..capacity], exit) catch |err| {
-                if (count > 0 and err == error.WouldBlock) {
-                    if (timer.read() > MAX_WAIT_NS) {
-                        break;
-                    }
-                }
+            const old_count = packet_batch.items.len;
+            packet_batch.items.len += recvMmsg(socket, packet_batch.unusedCapacitySlice(), exit) catch |err| {
```

Comment:
re-looking at this logic - i think we can remove a lot of unnecessary complexity - just set the timeout on the socket to something reasonable (given that we only want to break to check for exit signals) and simplify recvMmsg to fill up the batch until its full or it would block and loop again - wdyt - can huddle on it if you want

Reply:
To clarify on this, is this to mean that `recvMmsg` should become blocking?

Reply:
yeah 'blocking' by using a reasonable timeout (maybe 1second) - not full blocking

Reply:
I'd like to huddle on this, because I'm not exactly sure what the correct approach is.